graylog2-server/src/main/java/org/graylog2/rest/resources/system/UrlWhitelistResource.java

```diff
@@ -84,9 +89,27 @@ public Response put(@ApiParam(name = "whitelist", required = true) @NotNull fina
     @NoAuditEvent("Validation only")
     @Consumes(MediaType.APPLICATION_JSON)
     @RequiresPermissions(RestPermissions.URL_WHITELIST_READ)
-    public WhitelistCheckResponse check(@ApiParam(name = "url", required = true)
+    public WhitelistCheckResponse check(@ApiParam(name = "JSON body", required = true)
                              @Valid @NotNull final WhitelistCheckRequest checkRequest) {
         final boolean isWhitelisted = urlWhitelistService.isWhitelisted(checkRequest.url());
         return WhitelistCheckResponse.create(checkRequest.url(), isWhitelisted);
     }
+
+    @POST
+    @Path("/generate_regex")
+    @Timed
+    @ApiOperation(value = "Generates a regex that can be used as a value for a whitelist entry.")
+    @NoAuditEvent("Utility function only.")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public WhitelistRegexGenerationResponse generateRegex(@ApiParam(name = "JSON body", required = true)
+    @Valid @NotNull final WhitelistRegexGenerationRequest generationRequest) {
+        final String regex;
+        if (generationRequest.placeholder() == null) {
```

Comment:
Should we check here if the placeholder is empty as well? (e.g. by using `isNullOrEmpty(generationRequest.placeholder)`)

Reply:
Yep, makes sense. Fixed that.